CondFormats/HGCalObjects/interface/HGCalMappingModuleIndexer.h

```diff
@@ -73,6 +78,20 @@ class HGCalMappingModuleIndexer {
     globalTypesCounter_[typecodeIdx]++;
     globalTypesNErx_[typecodeIdx] = nerx;
     globalTypesNWords_[typecodeIdx] = nwords;
+
+    //add typecode string to map to retrieve fedId & modId later
+    if (typecode != "") {
+      if (typecodeMap_.find(typecode) != typecodeMap_.end()) {     // found key
+        const auto& [fedid_, modid_] = typecodeMap_.at(typecode);  // (fedId,modId)
+        edm::LogWarning("HGCalMappingModuleIndexer")
+            << "Found typecode " << typecode << " already in map (fedid,modid)=(" << fedid_ << "," << modid_
+            << ")! Overwriting with (" << fedid << "," << idx << ")...";
+      }
+      LogDebug("HGCalMappingModuleIndexer")
+          << "HGCalMappingModuleIndexer::processNewModule: Adding typecode=\"" << typecode << "\" with fedid=" << fedid
+          << ", idx=" << idx << " (will be re-indexed after finalize)";
+      typecodeMap_[typecode] = std::make_pair(fedid, idx);
+    }
```

Comment:
I'd suggest to move the implementation of all these heavy functions to a `.cc` file. Long functions will likely not be inlined anyway, they will likely increase the binary sizes, and you have to recompile all dependent code when doing changes.

Reply:
Done in commit https://github.com/cms-sw/cmssw/pull/45665/commits/b7641b776fb15fc28b2c72fce1637a7479905792#diff-7af13efc9c8801a946e6ab0c7439e1458229d32c1bf1493b1935f35b8e0e9059.